File: queue.js

var MyQueue = function() {
    this.queue = [];
    this.index = 0;
};

/** 
 * @param {number} x
 * @return {void}
 */
MyQueue.prototype.push = function(x) {
    this.queue.push(x);

};

/**
 * @return {number}
 */
MyQueue.prototype.pop = function() {
    this.index ++;
    return this.queue[this.index-1];
};

/**
 * @return {number}
 */
MyQueue.prototype.peek = function() {
    return this.queue[this.index];
};

/**
 * @return {boolean}
 */
MyQueue.prototype.empty = function() {
    if (this.index == this.queue.length){
        return true;
    }
    else{
        return false;
    }
};

File: queue.py
class MyQueue:

    def __init__(self):
        self.queue = []
        self.index = 0

    def push(self, x: int) -> None:
        self.queue.append(x)

    def pop(self) -> int:
        self.index += 1
        return self.queue[self.index-1]

    def peek(self) -> int:
        return self.queue[self.index] 

    def empty(self) -> bool:
        if self.index == len(self.queue):
            return True
        else:
            return False
    
    def __str__(self):
        if str.queue[self.index:]:
            return str(self.queue[self.index:])
        else:
            return 
